frontend/packages/dev-console/src/components/topology2/TopologyPage.tsx

```diff
@@ -86,16 +87,31 @@ const TopologyPage: React.FC<Props> = ({
         variant={showListView ? NamespacedPageVariants.light : NamespacedPageVariants.default}
         hideApplications={showListView}
         toolbar={
-          <Tooltip position="left" content={showListView ? 'Topology View' : 'List View'}>
-            <Link
-              className="pf-c-button pf-m-plain"
-              to={`/topology/${namespace ? `ns/${namespace}` : 'all-namespaces'}${
-                showListView ? '' : '/list'
-              }`}
-            >
-              {showListView ? <TopologyIcon size="md" /> : <ListIcon size="md" />}
-            </Link>
-          </Tooltip>
+          <div style={{ display: 'flex' }}>
+            {!showListView && (
+              <Popover
+                aria-label="Shortcuts"
+                bodyContent={TopologyShortcuts}
+                position="left"
+                maxWidth="50rem"
+              >
+                <Button type="button" variant="link" className="odc-topology__shortcuts-button">
+                  <QuestionCircleIcon className="co-icon-space-r co-p-has-sidebar__sidebar-link-icon" />
+                  View Shortcuts
+                </Button>
+              </Popover>
```

Comment:
I suspect this whole thing should be a component in the `console-shared` package. #3178 already makes use of something like this and #3496 provides the common interface. I think we can composite these together and make a single "ShortcutHelper" that takes a fragment or list of `Shortcut`s

Reply:
So you are suggesting to create a generic wrapper for this button-popover pair ? I am not sure how much value it adds because anyways we will have to pass all the props as they can be different for different usecases. Also only button icon and style is going to be similar. @christianvogt wdyt?

Reply:
I can see the way we surface shortcuts differing.
There are some slight differences at the moment.
I suggest a tech debt item where we can put a bit more thought into alignment after 4.3.

Reply:
I think this is literally the basis of React. Composition of components is what we do everywhere. DRY is a strong supported principle of React.

Definitely seeing something like this:
```
export interface UserShortcutsProps {
  shortcuts: React.ReactNode;
}

const UserShortcuts: React.FC<UserShortcutsProps> = ({ shortcuts, ...props }) => {
  return (
    <Popover
      aria-label="Shortcuts"
      bodyContent={<ShortcutTable>{shortcuts}</ShortcutTable>}
      {...props}
    >
      <Button type="button" variant="link" isInline>
        <QuestionCircleIcon className="co-icon-space-r co-p-has-sidebar__sidebar-link-icon" />
        Shortcuts
      </Button>
    </Popover>
  );
};

render(
  <UserShortcuts
    shortcuts={
      <>
        <Shortcut drag>move node or application</Shortcut>
        <Shortcut shift drag>
          move node in/out of an application group
        </Shortcut>
        <Shortcut rightClick>access respective context menus</Shortcut>
        <Shortcut click>view node or application details</Shortcut>
        <Shortcut hover>show node drag and drop connector</Shortcut>
      </>
    }
  />,
  document.getElementById('example'),
);
```

Allows very easy use of this paradigm. We want a set of `UserShortcuts` in the same design and structure, why not have a component that declares this?

Reply:
I'm all for creating reusable components.
if you look at the yaml editor shortcuts, they specifically made the menu a certain distance from the button and it opens down.
![image](https://user-images.githubusercontent.com/14068621/69269137-65dde280-0b9e-11ea-8f0f-f50995237bba.png)

There is a slight inconsistency being introduced with these two designs. I just didn't want to get into it at this point as we close off on 4.3.
If we can get this PR in, then we can align afterwards.
Similar to the other PR getting in and us creating the reusable shortcuts component.

Reply:
> There is a slight inconsistency being introduced with these two designs. I just didn't want to get into it at this point as we close off on 4.3.

If we log a tech debt ticket and address it in the next release, I'm okay with not doing it now. I just don't believe duplication of code is ever needed... especially not in this case. The component I wrote out (and the one we should implement) should be able to take additional properties to customize the `Popover` - so matching existing designs shouldn't be an issue. Not all shortcut buttons should exist with a down or left popover... I feel that is the page's choice.

However, I do agree, making this change now is probably not productive to closing off a release.

Reply:
I agree that we should make reusable components wherever possible and I have always tried to minimize the duplication of code as much as possible but here the usecases have totally different popover properties and text. If only one or two things were different we could have pulled out the common bits and made a generic component that can be customized according to the usecase passing in the variable props. I think creating a tech debt will be the best idea to think upon how do we handle this in the best possible way.